osu.Game/Rulesets/Objects/Drawables/DrawableHitObject.cs

```diff
@@ -82,32 +83,41 @@ protected DrawableHitObject(HitObject hitObject)
             HitObject = hitObject;
         }
 
+        private readonly Bindable<Skin> skin = new Bindable<Skin>();
+
         [BackgroundDependencyLoader]
-        private void load(AudioManager audio)
+        private void load(AudioManager audio, SkinManager skins)
         {
             var samples = GetSamples();
             if (samples.Any())
             {
                 if (HitObject.SampleControlPoint == null)
                     throw new ArgumentNullException(nameof(HitObject.SampleControlPoint), $"{nameof(HitObject)}s must always have an attached {nameof(HitObject.SampleControlPoint)}."
                                                                                           + $" This is an indication that {nameof(HitObject.ApplyDefaults)} has not been invoked on {this}.");
-
-                foreach (SampleInfo s in samples)
+                void loadSamples(Skin skin)
                 {
-                    SampleInfo localSampleInfo = new SampleInfo
+                    Samples.Clear();
+
+                    foreach (SampleInfo s in samples)
                     {
-                        Bank = s.Bank ?? HitObject.SampleControlPoint.SampleBank,
-                        Name = s.Name,
-                        Volume = s.Volume > 0 ? s.Volume : HitObject.SampleControlPoint.SampleVolume
-                    };
+                        SampleInfo localSampleInfo = new SampleInfo
+                        {
+                            Bank = s.Bank ?? HitObject.SampleControlPoint.SampleBank,
+                            Name = s.Name,
+                            Volume = s.Volume > 0 ? s.Volume : HitObject.SampleControlPoint.SampleVolume
+                        };
 
-                    SampleChannel channel = localSampleInfo.GetChannel(audio.Sample, SampleNamespace);
 
-                    if (channel == null)
-                        continue;
+                        SampleChannel channel = localSampleInfo.GetChannel(skin.GetSample, SampleNamespace) ?? localSampleInfo.GetChannel(audio.Sample.Get, SampleNamespace);
 
-                    Samples.Add(channel);
+                        if (channel == null) return;
+
+                        Samples.Add(channel);
+                    }
                 }
+
+                skin.ValueChanged += loadSamples;
+                skin.BindTo(skins.CurrentSkin);

```

Comment:
No way, I thought you were going to abstract this out and have SkinManager return you the bindables... We shouldn't be storing local skin bindables like this.

Reply:
temporary implementation

Reply:
wait, not temporary. this is intentional. i don't want the SkinManager to be handling anything but skins.

Reply:
As discussed, let's add a SkinnableSample to handle this logic for us, in a similar way to SkinnableDrawable.